Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/EventTag/ByEventTag.tsx

```diff
@@ -0,0 +1,166 @@
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/31/2023 - Parker Dinsdale
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { EventTagSlice } from '../Store/Store';
+
+import Table from '@gpa-gemstone/react-table'
+import { OpenXDA, Application } from '@gpa-gemstone/application-typings';
+import { SearchBar, Search, Modal } from '@gpa-gemstone/react-interactive';
+import { CrossMark, HeavyCheckMark } from '@gpa-gemstone/gpa-symbols';
+import EventTagForm from './EventTagForm';
+
+
+const EventTags: Application.Types.iByComponent = (props) => {
+    const dispatch = useAppDispatch();
+
+    const data = useAppSelector(EventTagSlice.SearchResults);
+    const status = useAppSelector(EventTagSlice.SearchStatus);
+
+    const allTags = useAppSelector(EventTagSlice.Data);
+    const allTagsStatus = useAppSelector(EventTagSlice.Status);
+
+    const [mode, setMode] = React.useState<'View'|'Add'|'Edit'>('View');
+    const [sortField, setSortField] = React.useState<keyof OpenXDA.Types.EventTag>('Name');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [errors, setErrors] = React.useState<string[]>([]);
+
+    const emptyRecord = { ID: 0, Name: '', Description: '', ShowInFilter: false };
+
+    const EventTagSearchFields = [
+        { label: 'Name', key: 'Name', type: 'string', isPivotField: false },
+        { label: 'Description', key: 'Description', type: 'string', isPivotField: false },
+    ];
+    const EventTagDefaultSearchField = { label: 'Name', key: 'Name', type: 'string', isPivotField: false };
+    const [search, setSearch] = React.useState<Array<Search.IFilter<OpenXDA.Types.EventTag>>>([]);
+
+    const [record, setRecord] = React.useState<OpenXDA.Types.EventTag>(emptyRecord);
+
+    React.useEffect(() => {
+        if (status === 'unintiated' || status === 'changed')
+            dispatch(EventTagSlice.DBSearch({ filter: search, sortField, ascending }));
+    }, [status]);
+
+    React.useEffect(() => {
+        dispatch(EventTagSlice.DBSearch({ filter: search, sortField, ascending }));
```

Comment:
you can use `dispatch(EventTagSlice.DBSearch({ filter: search}));` instead of `setSearch` and `dispatch(EventTagSlice.Sort({sortField,ascending}))`  because `GenericSlice` guarantees it's `sortField`,`ascewnding`, `filter` parameters are matching what `data` is sorted and filtered by....